src/lib/inbox/inbox.ts

```diff
@@ -81,6 +93,32 @@ export class InboxTools {
     )
   }
 
+  /**
+   * We should use nodeInterface to get the gas estimate is because we
+   * are making a delayed inbox message which doesn't need l1 calldata
+   * gas fee part.
+   */
+  private async estimateGasWithoutL1Part(
```

Comment:
It would probably be more general to return the full result of calling `gasEstimateComponents` then do the subtraction where needed. You could also return the full result and populate an extra property on it `gasEstimateForL2`?